classes/interaction.lua

```diff
@@ -29,6 +29,13 @@ function Interaction:constructor(data)
     self.options = data.options
     self.isDestroyed = false
     self.DuiOptions = {}
+    self.sprite = data.sprite
+
+    if data.sprite ~= nil then
```

Comment:
for here we can use the optional chaining operator `?` to make sure that data.sprite AND data.sprite.dict exist.

if data?.sprite?.dict then

Reply:
Switched to chaining operator in commit [f1c7f93](https://github.com/Sleepless-Development/sleepless_interact/pull/7/commits/f1c7f9380bf5f94cd95dcd7aba6c1c55d00c2f44)